test/indexer/test_tdr.py

```diff
@@ -0,0 +1,211 @@
+    attrgetter,
+    itemgetter,
+)
+from typing import (
+    Mapping,
+)
+import unittest
+
+from furl import (
+    furl,
+)
+from more_itertools import (
+    first,
+    one,
+)
+from tinyquery import (
+    tinyquery,
+)
+
+from azul import (
+    RequirementError,
+    cached_property,
+    config,
+    lru_cache,
+)
+from azul.bigquery import (
+    BigQueryRow,
+    BigQueryRows,
+)
+from azul.indexer import (
+    BundleFQID,
+)
+from azul.plugins.repository import (
+    tdr,
+)
+from azul.plugins.repository.tdr import (
+    TDRBundle,
+)
+from azul.terra import (
+    TDRSource,
+)
+from azul.types import (
+    JSONs,
+)
+from indexer import (
+    CannedBundleTestCase,
+)
+
+snapshot_id = 'cafebabe-feed-4bad-dead-beaf8badf00d'
+bundle_uuid = '1b6d8348-d6e9-406a-aa6a-7ee886e52bf9'
+
+
+class TestTDRPlugin(CannedBundleTestCase):
+    test_source = TDRSource(project='test_project', name='snapshot', is_snapshot=True)
+
+    @cached_property
+    def tinyquery(self) -> tinyquery.TinyQuery:
+        return tinyquery.TinyQuery()
+
+    def test_list_links_ids(self):
+        source = self._test_source(is_snapshot=True)
+        current_version = '2001-01-01T00:00:00.000001Z'
+        links_ids = ['42-abc', '42-def', '42-ghi', '86-xyz']
+        self._make_mock_entity_table(source=source,
+                                     table_name='links',
+                                     rows=[
+                                         dict(links_id=links_id, version=current_version, content='{}')
+                                         for links_id in links_ids
+                                     ])
+        plugin = TestPlugin(source, self.tinyquery)
+        bundle_ids = plugin.list_links_ids(prefix='42')
+        bundle_ids.sort(key=attrgetter('uuid'))
+        self.assertEqual(bundle_ids, [
+            BundleFQID('42-abc', current_version),
+            BundleFQID('42-def', current_version),
+            BundleFQID('42-ghi', current_version)
+        ])
+
+    def _test_source(self, *, is_snapshot: bool) -> TDRSource:
+        return TDRSource(project='foo', name='bar', is_snapshot=is_snapshot)
+
+    @lru_cache
+    def _canned_bundle(self, source: TDRSource) -> TDRBundle:
+        fqid = BundleFQID(bundle_uuid, '')
+        canned_result = self._load_canned_file(fqid, 'result.tdr')
+        manifest, metadata = canned_result['manifest'], canned_result['metadata']
+        return TDRBundle(source=source,
+                         uuid=bundle_uuid,
+                         version=one(e['version'] for e in manifest if e['name'] == 'links.json'),
+                         manifest=manifest,
+                         metadata_files=metadata)
+
+    def _make_mock_tdr_tables(self,
+                              source: TDRSource,
+                              bundle_fqid: BundleFQID,
+                              *,
+                              tables: Mapping[str, JSONs] = None) -> None:
+        if tables is None:
```

Comment:
There is only one call site AFAICT, and it doesn't pass `tables` so no need for this conditional or the argument.

Reply:
Look for other cases of unnecessarily complex method/fucntion signatures.